afids_utils/io.py

```diff
@@ -0,0 +1,85 @@
+"""Methods for loading and saving nifti files"""
+from __future__ import annotations
+
+import csv
+from os import PathLike
+
+import nibabel as nib
+import numpy as np
+import pandas as pd
+from numpy.typing import NDArray
+
+from afids_utils.exceptions import InvalidFiducialNumberError
+
+AFIDS_FIELDNAMES = [
+    "id",
+    "x",
+    "y",
+    "z",
+    "ow",
+    "ox",
+    "oy",
+    "oz",
+    "vis",
+    "sel",
+    "lock",
+    "label",
+    "desc",
+    "associatedNodeID",
+]
+
+
+def load_nii(
+    nii_path: PathLike[str] | str,
+) -> tuple[NDArray[np.single], NDArray[np.single]]:
+    """Load and normalize nifti data and scanner-to-world transform"""
+    # Load unnormalized nifti volume and transform
+    nii = nib.loadsave.load(nii_path)
+    nii_affine = nii.affine.astype(np.single)
+    nii_data = nii.get_fdata().astype(np.single)
+    # Normalize nifti volume
+    nii_data = (nii_data - nii_data.min()) / (nii_data.max() - nii_data.min())
+
+    return nii_affine, nii_data
+
+
+def get_afid(
+    fcsv_path: PathLike[str] | str, fid_num: int
+) -> NDArray[np.single]:
+    """Extract specific fiducial's spatial coordinates"""
+    if fid_num < 1 or fid_num > 32:
+        raise InvalidFiducialNumberError(fid_num)
+    fcsv_df = pd.read_csv(fcsv_path, sep=",", header=2)
+
+    return fcsv_df.loc[fid_num - 1, ["x", "y", "z"]].to_numpy(dtype="single")
+
+
+def afids_to_fcsv(
+    afid_coords: NDArray[np.single],
+    fcsv_template: PathLike[str] | str,
```

Comment:
I think we should get rid of this parameter and just always have a template in a `resources` directory we always load (or at least have that as a default).

Reply:
Can talk about this live, but I think these are the key two references for that conversation off the top of my head:
- https://python-poetry.org/docs/pyproject/#include-and-exclude
- https://docs.python.org/3/library/importlib.resources.html